lib/rucio/core/trace.py

```diff
@@ -206,6 +207,25 @@
     'sm_get_a': GET_SCHEMA
 }
 
+FORMAT_CHECKER = draft7_format_checker
+
+
+@FORMAT_CHECKER.checks(format="ipv4_or_ipv6")
+def ip_format_checker(value: str) -> bool:
+    """
+    Validates IPv4 or IPv6 string values. json schemas can use `ipv4_or_ipv6` as a valid `format` argument
+    """
+    ipv4_checker = FORMAT_CHECKER.checkers['ipv4'][0]
+    ipv6_checker = FORMAT_CHECKER.checkers['ipv6'][0]
+    try:
+        result = ipv4_checker(value) or ipv6_checker(value)
+    except (OSError, ValueError, struct.error):
```

Comment:
hmm, I didn't check at all, but the exception seems quite stangely placed. 
if ipv4_checker raises a valueError on  ipv6 addresses, it will fail the validation. Isn't this the case ?

Reply:
Made it cleaner now ;)